targets/atmega2560.json

```diff
@@ -0,0 +1,20 @@
+{
+    "inherits": ["avr"],
+    "llvm-target": "avr-atmel-none",
+    "cpu": "atmega2560",
+    "build-tags": ["atmega2560"],
+    "cflags": [
+        "-mmcu=atmega2560"
+    ],
+    "ldflags": [
+        "-mmcu=avr6",
+        "-Wl,--defsym=_bootloader_size=0",
```

Comment:
I looked at the Arduino IDE and it appears that the bootloader is 8kB in size (253952 - 256kB = 8kB):

    mega.menu.cpu.atmega2560.upload.maximum_size=253952

Reply:
> Thank you for this contribution!
> 
> This PR should be rebased on top of the dev branch after #946 has been merged.

Of course this will be done.

Reply:
The comment grouping on Github is being funny. 

I'll make the bootloader change.